packages/clerk-js/src/ui/components/ApiKeys/ApiKeys.tsx

```diff
@@ -0,0 +1,170 @@
+import { isClerkAPIResponseError } from '@clerk/shared/error';
+import { useClerk, useOrganization, useUser } from '@clerk/shared/react';
+import type { CreateAPIKeyParams } from '@clerk/types';
+import { lazy, useState } from 'react';
+import useSWRMutation from 'swr/mutation';
+
+import { useApiKeysContext, withCoreUserGuard } from '../../contexts';
+import { Box, Button, Col, Flex, Flow, Icon, localizationKeys, useLocalizations } from '../../customizables';
+import { InputWithIcon, Pagination, useCardState, withCardStateProvider } from '../../elements';
+import { Action } from '../../elements/Action';
+import { MagnifyingGlass } from '../../icons';
+import { mqu } from '../../styledSystem';
+import { ApiKeysTable } from './ApiKeysTable';
+import type { OnCreateParams } from './CreateApiKeyForm';
+import { CreateApiKeyForm } from './CreateApiKeyForm';
+import { useApiKeys } from './useApiKeys';
+
+type APIKeysPageProps = {
+  subject: string;
+  perPage?: number;
+  revokeModalRoot?: React.MutableRefObject<HTMLElement | null>;
+};
+
+const RevokeAPIKeyConfirmationModal = lazy(() =>
+  import(/* webpackChunkName: "revoke-api-key-modal"*/ './RevokeAPIKeyConfirmationModal').then(module => ({
+    default: module.RevokeAPIKeyConfirmationModal,
+  })),
+);
+
+export const APIKeysPage = ({ subject, perPage, revokeModalRoot }: APIKeysPageProps) => {
+  const {
+    apiKeys,
+    isLoading,
+    search,
+    setSearch,
+    page,
+    setPage,
+    pageCount,
+    itemCount,
+    startingRow,
+    endingRow,
+    cacheKey,
+  } = useApiKeys({ subject, perPage });
+  const card = useCardState();
+  const { trigger: createApiKey, isMutating } = useSWRMutation(cacheKey, (_, { arg }: { arg: CreateAPIKeyParams }) =>
+    clerk.createApiKey(arg),
+  );
+  const { t } = useLocalizations();
+  const clerk = useClerk();
+  const [isRevokeModalOpen, setIsRevokeModalOpen] = useState(false);
+  const [selectedApiKeyId, setSelectedApiKeyId] = useState('');
+  const [selectedApiKeyName, setSelectedApiKeyName] = useState('');
+
+  const handleCreateApiKey = async (params: OnCreateParams, closeCardFn: () => void) => {
+    try {
+      await createApiKey(params);
+      closeCardFn();
+      card.setError(undefined);
+    } catch (err: any) {
+      if (isClerkAPIResponseError(err)) {
+        if (err.status === 409) {
+          card.setError('API Key name already exists');
+        }
+      }
+    }
+  };
+
+  const handleRevoke = (apiKeyId: string, apiKeyName: string) => {
+    setSelectedApiKeyId(apiKeyId);
+    setSelectedApiKeyName(apiKeyName);
+    setIsRevokeModalOpen(true);
+  };
+
+  return (
+    <Col
```

Comment:
Across the component, we are missing descriptors to allow modifying this specific component. 

You can see in the following screenshot, there is a good amount of only internal classes. Which will make it difficult to style.

![Screenshot 2025-05-27 at 9 32 58 AM](https://github.com/user-attachments/assets/b3026666-719e-47cc-ba42-109dd8af9f84)

Reply:
Will do this in another PR - https://github.com/clerk/javascript/pull/6095